### PR TITLE
fix: for readiness check both the phase and ready status

### PR DESCRIPTION
**Pull request by deepthidevaki**
Observed that the ready status = true when the pod is terminating.

### CODE REVIEW

**Review by Zelldon — COMMENTED** [review]
I would feel better if we have a test for that. Could you take a look at this?

Like adding a test where you setup a pod, with pod status Running, Failed etc. and verify the result of that check function?


Here you find the functions which creates the pod https://github.com/zeebe-io/zeebe-chaos/blob/main/go-chaos/internal/helper_test.go#L73-L80 in the fake k8 environment

Here https://github.com/zeebe-io/zeebe-chaos/blob/main/go-chaos/internal/pods_test.go I would see the test like

* Test_ShouldReturnTrueWhenBrokerPodsAreReady
* Test_ShouldReturnFalseWhenOneBrokerPodIsNotReady

**Comment by deepthidevaki** [comment]
@Zelldon Do you know how I can mock a respose for "get pods" to return ready or not ready pods? When creating a pod using the fakeclient we are only providing the spec of the pod. We cannot set the status.

**Comment by Zelldon** [comment]
@deepthidevaki you can define the status here in the struct :) https://github.com/zeebe-io/zeebe-chaos/blob/main/go-chaos/internal/helper_test.go#L73-L80 Maybe create another function which creates the pod with a status.

**Comment by deepthidevaki** [comment]
> @deepthidevaki you can define the status here in the struct :) https://github.com/zeebe-io/zeebe-chaos/blob/main/go-chaos/internal/helper_test.go#L73-L80 Maybe create another function which creates the pod with a status.

Aah, Right. I tried to add it in the Spec :see_no_evil:

File: go-chaos/internal/pods.go [hunk 1]
```diff
@@ -121,32 +121,33 @@ func (c K8Client) RestartPodWithGracePeriod(podName string, gracePeriodSec *int6
...
-		gatewaysAreRunning, err := c.checkIfGatewaysAreRunning()
-		if err != nil {
-			return err
-		}
+	// Keep checking until we're timed out
+	for {
+		select {
+		case <-timedOut:
+			return errors.New(fmt.Sprintf("Awaited readiness of pods in namespace %v, but timed out after %v", c.GetCurrentNamespace(), timeout))
+		case <-ticker:
```
Comment (Zelldon) [inline go-chaos/internal/pods.go:136]: :+1: Yeah I thought also last week that we can write this better with channels thanks for improving this :rocket:

File: go-chaos/internal/pods_test.go [hunk 2]
```diff
@@ -282,8 +283,50 @@ func Test_ShouldReturnTrueWhenBrokersAreReady(t *testing.T) {
...
+	require.NoError(t, err)
+	k8Client.CreateBrokerPodsWithStatus(t, selector, "broker1", v1.PodRunning, true)
+	k8Client.CreateBrokerPodsWithStatus(t, selector, "broker2", v1.PodRunning, false)
+
+	gatewaySelector, err := metav1.ParseToLabelSelector(getSelfManagedGatewayLabels())
+	require.NoError(t, err)
+	k8Client.CreateDeploymentWithLabelsAndName(t, gatewaySelector, "gateway")
+
+	// when
+	err = k8Client.AwaitReadinessWithTimeout(2 * time.Second)
```
Comment (Zelldon) [inline go-chaos/internal/pods_test.go:307]: :wrench: Could we reduce the timeout otherwise the tests take a bit long

**Review by Zelldon — APPROVED** [review]
Thank you very much for adding additional tests @deepthidevaki :+1: :rocket: 

Please have a look at my comments, maybe you can do that as follow up. I will merge it anyway, since I'm currently testing the QA and this helps here.

**Pull request merged**